### PR TITLE
Add QueryableExtensions and associated tests

### LINKED ISSUE #15: Implement IQueryable Extensions
After running SQL Profiler, we noticed that using the `IEnumerable.IsNullOrEmpty()` extension method in this repo will run a full `select` on a table and _then_ evaluate whether or not there are elements in memory, vs. the `IQueryable.Any()` which modifies the `select` like so:

```
exec sp_executesql N'SELECT CASE
    WHEN EXISTS (
        SELECT 1
        FROM [UserRoles] AS [e]
        WHERE ([e].[DeletedOn] IS NULL AND ([e].[UserId] = @__userId_0)) AND ([e].[EulaAccepted] = 0))
    THEN CAST(1 AS BIT) ELSE CAST(0 AS BIT)
END',N'@__userId_0 bigint',@__userId_0=4
```

At the very least, we should implement an `IsEmpty()`, and `IsNullOrEmpty()` for `IQueryable` (which call the `IQueryable.Any()` method), to have parity with `IEnumerable`. I imagine there are many places where we're checking `IsNullOrEmpty()` on an `IQueryable` `ResultObject` and it is doing that check in memory instead of in the query.

I'd be open to other suggestions for extensions as well.

### PR DESCRIPTION
**Pull request by brandongregoryscott**
Closes #15 

Based off the existing `EnumerableExtensions`

### CODE REVIEW

File: src/AndcultureCode.CSharp.Extensions/QueryableExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace AndcultureCode.CSharp.Extensions
+{
+    public static class QueryableExtensions
```
Comment (wintondeshong) [inline src/AndcultureCode.CSharp.Extensions/QueryableExtensions.cs:7]: I just noticed we broken from following exact types in extension file names.

Believe this should be `IQueryableExtensions` and EnumerableExtensions should be updated to `IEnumerableExtensions`

Comment (brandongregoryscott) [reply src/AndcultureCode.CSharp.Extensions/QueryableExtensions.cs:7]: I noticed this as well, but followed the existing pattern. Can definitely rename this!

**Comment by brandongregoryscott** [comment]
> @brandongregoryscott I'm not sure the last time I even thought about this scenario, how does this play out from an implementation perspective? I'm assuming being IQueryable inherits from IEnumerable and these extensions are both in the same namespace, it picks the closest/exact (in this case) match of typed extensions. IQueryable.XYZ vs. IEnumerable.XYZ. Just checking.

I'm not sure sure how often I encounter this either - but you're correct, it does inherit from IEnumerable and based on what intellisense is telling me, it is smart enough to pick the lowest child's extensions based on type:

image

**Comment by wintondeshong** [comment]
> > @brandongregoryscott I'm not sure the last time I even thought about this scenario, how does this play out from an implementation perspective? I'm assuming being IQueryable inherits from IEnumerable and these extensions are both in the same namespace, it picks the closest/exact (in this case) match of typed extensions. IQueryable.XYZ vs. IEnumerable.XYZ. Just checking.
> 
> I'm not sure sure how often I encounter this either - but you're correct, it does inherit from IEnumerable and based on what intellisense is telling me, it is smart enough to pick the lowest child's extensions based on type:
> 
> image

Perfect! What I assumed, but just double checking. Thanks for checking

**Pull request merged**